gateway/src/shard/error.rs

```diff
@@ -122,6 +136,11 @@ impl Display for Error {
                 value
             ),
             Self::ParsingUrl { url, .. } => write!(f, "The gateway URL {:?} is invalid", url),
+            Self::PayloadInvalid { .. } => write!(
+                f,
+                "The binary payload received from Discord wasn't UTF-8 valid"
```

Comment:
Should this not be another error message, something along the lines that the payload received was invalid. not nessecarily that the utf-8 encoding was the issue.